### PR TITLE
Geant4 is required to build the full physics demo app

### PR DESCRIPTION
**Pull request by mrguilima**
I could not build the full physics demo in my area, because I don't enable Geant4.  
So I have added a Geant4=ON requirement for this demo app.

### CODE REVIEW

**Review by stognini — APPROVED** [review]

File: app/CMakeLists.txt [hunk 1]
```diff
@@ -189,7 +189,7 @@ endif()
 # DEMO: full physics loop
 #-----------------------------------------------------------------------------#
 
-if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_VecGeom)
+if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_VecGeom AND CELERITAS_USE_Geant4)
```
Comment (sethrj) [inline app/CMakeLists.txt:192]: Per the slack discussion -- the app itself will build fine without geant4, it's just the testing that requires it (and also ROOT) to generate and use the data. Can you revert this line, then add restrictions to the test itself?

**Review by sethrj — CHANGES_REQUESTED** [review]

File: app/CMakeLists.txt (hunk 1 above)
Comment (mrguilima) [reply app/CMakeLists.txt]: Done.

File: app/CMakeLists.txt [hunk 2]
```diff
@@ -225,7 +225,7 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_VecGeom)
   )
 
   # TODO: update input files and enable test
-  if(CELERITAS_BUILD_TESTS)
+  if(CELERITAS_BUILD_TESTS AND CELERITAS_USE_Geant4)
```
Comment (sethrj) [inline app/CMakeLists.txt:228]: ` AND CELERITAS_USE_ROOT` as well

Comment (mrguilima) [reply app/CMakeLists.txt]: Oops... sorry, this is also done now.

**Review by sethrj — APPROVED** [review]

**Pull request merged**